frontend/ui/widget/keyboardlayoutdialog.lua

```diff
@@ -0,0 +1,179 @@
+--[[--
+This widget displays an keyboard layout dialog.
+]]
+
+local Blitbuffer = require("ffi/blitbuffer")
+local ButtonTable = require("ui/widget/buttontable")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local Font = require("ui/font")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local Geom = require("ui/geometry")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local Language = require("ui/language")
+local LineWidget = require("ui/widget/linewidget")
+local RadioButtonTable = require("ui/widget/radiobuttontable")
+local Size = require("ui/size")
+local TextWidget = require("ui/widget/textwidget")
+local UIManager = require("ui/uimanager")
+local VerticalGroup = require("ui/widget/verticalgroup")
+local VerticalSpan = require("ui/widget/verticalspan")
+local _ = require("gettext")
+local Screen = require("device").screen
+local orderedPairs = require("ffi/util").orderedPairs
+
+local KeyboardLayoutDialog = InputContainer:new{
+    is_always_active = true,
+    title = _("Keyboard Layout"),
+    modal = true,
+    width = Screen:getWidth() * 0.8,
+    face = Font:getFace("cfont", 22),
+
+    title_face = Font:getFace("x_smalltfont"),
+
+    title_padding = Size.padding.default,
+    title_margin = Size.margin.title,
+    button_padding = Size.padding.default,
+    border_size = Size.border.window,
+
+}
+
+
+function KeyboardLayoutDialog:init()
+    -- Title & description
+    self.title_widget = FrameContainer:new{
+        padding = self.title_padding,
+        margin = self.title_margin,
+        bordersize = 0,
+        TextWidget:new{
+            text = self.title,
+            face = self.title_face,
+            max_width = self.width,
+        }
+    }
+    self.title_bar = LineWidget:new{
+        dimen = Geom:new{
+            w = self.width,
+            h = Size.line.thick,
+        }
+    }
+
+    local buttons = {}
+    local radio_buttons = {}
+    for k, _ in orderedPairs(self.parent.keyboard.lang_to_keyboard_layout) do
+        table.insert(radio_buttons, {
+            {
+            text = Language:getLanguageName(k),
+            checked = self.parent.keyboard:getKeyboardLayout() == k,
+            provider = k,
+            },
+        })
+    end
+
+    table.insert(buttons, {
+        {
+            text = _("Cancel"),
+            callback = function()
+                UIManager:close(self.parent.keyboard_layout_dialog)
+            end,
+        },
+        {
+            text = _("Chose Language"),
```

Comment:
I'd say something like:
```suggestion
            text = _("Switch to layout"),
```

(The text as written should read `Choose layout`, but that aside.)